plugins/catalog-react/src/filters.ts

```diff
@@ -185,8 +200,66 @@ export class EntityNamespaceFilter implements EntityFilter {
   }
 }
 
+/**
+ * @public
+ */
+export class EntityUserListFilter implements EntityFilter {
+  private constructor(
+    readonly value: UserListFilterKind,
+    readonly refs?: string[],
+  ) {}
+
+  static owned(ownershipEntityRefs: string[]) {
+    return new EntityUserListFilter('owned', ownershipEntityRefs);
+  }
+
+  static all() {
+    return new EntityUserListFilter('all');
+  }
+
+  static starred(starredEntityRefs: string[]) {
+    return new EntityUserListFilter('starred', starredEntityRefs);
+  }
+
+  getCatalogFilters(): Record<string, string[]> {
+    if (this.value === 'owned') {
+      return { 'relations.ownedBy': this.refs ?? [] };
+    }
+    if (this.value === 'starred') {
+      return {
+        'metadata.name': this.refs?.map(e => parseEntityRef(e).name) ?? [],
```

Comment:
actually this works, as the possible extra entities are filtered out locally by `filterEntity` below 👇

but your suggestion might simplify things a bit, let me check

Reply:
so I've looked into it and it looks like the function `getCatalogFilters` of `EntityFilter` is not expecting to return an array, but only `Record<string, string | symbol | (string | symbol)[]>`.
I think this is due to that filters are combined all together as AND, so for example: `EntityLifecycleFilter`, `EntityTagFilter` returns two different filters which are combined together into a single filter:

```js
 {
   'metadata.tags': ['tag1, 'tag2'],
   'spec.lifecycle': 'production' 
 }
 ```
 
so I don't think I could really combine multiple OR rules listing kind, name and namespace of the starred entities together with the combined filter coming from other filters, as it will require some nested filters, which don't seem to be supported in the API. I'm sure catalog can do that, but not through the rest api